### PR TITLE
Fix JsonStdio transport not to exit as soon as it started.

### PR DESCRIPTION
**Pull request by arrowd**

### CODE REVIEW

File: src/Haskell/Ide/Engine/Transport/JsonStdio.hs [hunk 1]
```diff
@@ -79,6 +79,7 @@ run dispatcherProc cin = flip E.catches handlers $ do
     _opid <- forkIO $ outWriter rout
     _rpid <- forkIO $ reactor rout
     dispatcherProc dispatcherEnv
+    forever $ threadDelay maxBound
```
Comment (wz1000) [inline src/Haskell/Ide/Engine/Transport/JsonStdio.hs]: how about
```haskell
dispatcherProc dispatcherEnv
race_ (outWriter rout) (reactor rout)
```

**Comment by wz1000** [comment]
See https://github.com/wz1000/haskell-ide-engine/commit/d2895ff34bc1947b50be80151875a6378f789387

**Comment by arrowd** [comment]
Whatever you think is better.

**Comment by wz1000** [comment]
take a look at #297

**Pull request closed without merging**